ci: add pushval action for building ModuleSite

jdrueckert opened the pull request:
- current assumption is that the deploy action does work with a mutable lockfile
  which is why the previously added build pushval actions do not reproduce failure we see no merging to master

jellysnake commented:
Let's preview this

gooey-splashsite commented:
Preview is available on http://preview.terasology.net/splashsite/pull/102 :rocket:

smsunarto commented:
Let's preview this

gooey-splashsite commented:
Preview is available on http://preview.terasology.net/splashsite/pull/102 :rocket:

bjorn commented:
Thanks. :-)

jdrueckert committed:
ci: add pushval action with mutable lockfile

- current assumption is that the deploy action does work with a mutable lockfile
  which is why the previously added build pushval actions do not reproduce failure we see no merging to master

jdrueckert committed:
chore: remove caching to avoid masking dependency issues

jdrueckert committed:
chore: utilize node 16 as the deploy action does

jdrueckert commented:
Now that finally reproduced the failure I see in the master-merge deploy action:
`error /home/runner/work/ModuleSite/ModuleSite/node_modules/node-sass: Command failed.`

This seems to be related to the `node-sass` module, on which I found the following: https://stackoverflow.com/questions/70281346/node-js-sass-version-7-0-0-is-incompatible-with-4-0-0-5-0-0-6-0-0, suggesting to replace `node-sass` with `sass`

jdrueckert committed:
fix: replace node-sass with sass dep

see https://stackoverflow.com/questions/70281346/node-js-sass-version-7-0-0-is-incompatible-with-4-0-0-5-0-0-6-0-0

jdrueckert committed:
chore: update dependencies
